cmd/snap/cmd_get.go

```diff
@@ -163,58 +225,12 @@ func (x *cmdGet) Execute(args []string) error {
 	}
 
 	isTerminal := terminal.IsTerminal(int(os.Stdin.Fd()))
-	rootRequested := len(confKeys) == 0
-	var confToPrint interface{} = conf
-
-	if rootRequested && len(conf) == 0 {
-		if !x.Document {
-			return fmt.Errorf("snap %q has no configuration", snapName)
-		}
-	} else {
-		if !x.Document && !x.List && len(confKeys) == 1 {
-			// if single key was requested, then just output the value unless it's a map,
-			// in which case it will be printed as a list below.
-			if _, ok := conf[confKeys[0]].(map[string]interface{}); !ok {
-				confToPrint = conf[confKeys[0]]
-			}
-		}
-
-		if cfg, ok := confToPrint.(map[string]interface{}); ok && !x.Document {
-			// TODO: remove this conditional and the warning below after a transition period.
-			if isTerminal || x.List {
-				w := tabWriter()
-				defer w.Flush()
-
-				fmt.Fprintf(w, "Key\tValue\n")
-				values := flattenConfig(cfg, rootRequested)
-				for _, v := range values {
-					fmt.Fprintf(w, "%s\t%v\n", v.Path, v.Value)
-				}
-				return nil
-			} else {
-				fmt.Fprintf(Stderr, i18n.G(`WARNING: The output of "snap get" will become a list with columns - use -d or -l to force the output format.\n`))
-			}
-		}
-
-		if x.Typed && confToPrint == nil {
-			fmt.Fprintln(Stdout, "null")
-			return nil
-		}
-
-		if s, ok := confToPrint.(string); ok && !x.Typed {
-			fmt.Fprintln(Stdout, s)
-			return nil
-		}
-	}
-
-	var bytes []byte
-	if confToPrint != nil {
-		bytes, err = json.MarshalIndent(confToPrint, "", "\t")
-		if err != nil {
-			return err
-		}
+	switch {
+	case x.Document:
+		return x.outputJson(conf)
+	case x.List || isTerminal:
```

Comment:
There's probably a test missing to catch this case in the old code:
```
if !x.Document && !x.List && len(confKeys) == 1 {
```
In other words, if we're printing out a single value and we didn't get an explicit request for it to be presented as a key/value doc or list, we still want this single value to be printed out individually (typed or not) so we can pick up the value up as in `$(snap get [-t] <snap> <key>)` etc.

Reply:
Sorry, that was a bad example as it won't be a terminal in this case, but I think we should still print out the individual value to give a hint of what the developer can expect when using the value. Otherwise people won't do `$(snap get ...)` because they won't realize the result would be different.

Reply:
Indeed, thanks for this. I updated the code and added a test, still not 100% certain I got it right will verify against the old code.

Reply:
I applied the following extra tests http://paste.ubuntu.com/25581934/ against the current master to double check that my new tests are still good, seems to be duplicating the old behaviour.